src/workers/generatePersonalizedDigestForUser.ts

```diff
@@ -0,0 +1,114 @@
+import { DataSource } from 'typeorm';
+import { getDiscussionLink, pickImageUrl, sendEmail } from '../common';
+import { ArticlePost, Source, User } from '../entity';
+import { UserPersonalizedDigest } from '../entity/UserPersonalizedDigest';
+import { messageToJson, Worker } from './worker';
+import { DayOfWeek } from '../types';
+import { MailDataRequired } from '@sendgrid/mail';
+import { nextDay } from 'date-fns';
+import { zonedTimeToUtc } from 'date-fns-tz';
+
+interface Data {
+  personalizedDigest: UserPersonalizedDigest;
+}
+
+const personalizedDigestPostsCount = 5;
+
+// TODO WT-1820 replace with feed config fetch
+const getMockedPosts = async ({
+  con,
+}: {
+  con: DataSource;
+}): Promise<(ArticlePost & { source: Source })[]> => {
+  const posts = await con.getRepository(ArticlePost).find({
+    take: personalizedDigestPostsCount,
+    order: {
+      createdAt: 'DESC',
+    },
+    relations: ['source'],
+  });
+
+  return Promise.all(
+    posts.map(async (post) => {
+      const source = await post.source;
+
+      return {
+        ...post,
+        source,
+      } as ArticlePost & { source: Source };
+    }),
+  );
+};
+
+const getEmailSendDate = ({
+  personalizedDigest,
+}: {
+  personalizedDigest: UserPersonalizedDigest;
+}): Date => {
+  const nextPreferredDay = nextDay(new Date(), personalizedDigest.preferredDay);
+  nextPreferredDay.setHours(personalizedDigest.preferredHour, 0, 0, 0);
+  const sendDateInPreferredTimezone = zonedTimeToUtc(
+    nextPreferredDay,
+    personalizedDigest.preferredTimezone,
+  );
+
+  return sendDateInPreferredTimezone;
+};
+
+const worker: Worker = {
+  subscription: 'api.generate-personalized-digest-for-user',
+  handler: async (message, con) => {
+    const data = messageToJson<Data>(message);
+    const { personalizedDigest } = data;
+
+    const user = await con.getRepository(User).findOneBy({
+      id: personalizedDigest.userId,
+    });
+
+    if (!user || !user.infoConfirmed) {
```

Comment:
We delete users who don't confirm in an hour so this is to make sure we don't send emails to those until they confirm.